src/oneshot/posix.rs

```diff
@@ -10,11 +10,18 @@ use core::{mem, ptr, time, task};
 use super::state::TimerState;
 use crate::alloc::boxed::Box;
 
-type RawFd = usize;
-
 mod ffi {
     use super::*;
 
+    #[derive(Clone, Copy)]
+    #[repr(C)]
+    pub struct timer_t(*mut libc::c_void);
```

Comment:
I think it uses `void*` though, not wrapper struct.
At least on Linux, if we're being pedantic about it.

So, ehm I don't really see much of benefit to just introduce this struct, as user is not supposed to be exposed to this timer_t

Reply:
`#[repr(C)]` for this wrapper struct makes it transparent, technically `#[repr(transparent)]` should be used, but `#[repr(C)]` also comes with `#[allow(non_camel_case_types)]`.

Alternatively it can be a type alias, but then we need to mark `State` with unsafe `Send` and `Sync` since the `*mut libc::c_void` bleeds through!

Reply:
yeah, that's why I used usize, because it is the same size (as it's size supposed to fit memory address).

I just don't feel like there is a need for extra lines, when usize could work 👀